extension/src/background/messageListener/lyraApiMessageListener.ts

```diff
@@ -65,18 +65,29 @@ export const lyraApiMessageListener = (
   };
 
   const submitTransaction = () => {
-    const { transactionXdr } = request;
+    let isDomainListedAllowed = false;
 
+    const { transactionXdr } = request;
     const transaction = StellarSdk.TransactionBuilder.fromXDR(
       transactionXdr,
       StellarSdk.Networks[NETWORK],
     );
 
     const { tab } = sender;
+    const tabUrl = tab?.url ? tab.url : "";
+
+    const sanitizedUrl = removeQueryParam(tabUrl);
```

Comment:
I actually really liked what you did with `punycode` in `popup`. Maybe we should create a helper method that takes a url and runs it through `getUrlHostname` and `punycode.toASCII` to get just the punycode domain and store that in the allowlist

That would also solve the issue of a different path on the same domain triggering the warning message again

Reply:
Updated @piyalbasu